extension/oauth2clientauthextension/extension.go

```diff
@@ -82,13 +91,24 @@ func (o *ClientCredentialsAuthenticator) Shutdown(_ context.Context) error {
 	return nil
 }
 
+func (ewts errorWrappingTokenSource) Token() (*oauth2.Token, error) {
```

Comment:
Would it make sense to add a test for this?

Reply:
I agree with @jpkrohling that we need a test with httptest server that micks the token endpoint with various codes and the corresponding validation against it. If you define  the error string as its own thing (as mentioned in my previous comment, you can validate with something like `assert.ErrorIs(t, err, errTokenSource)`

Reply:
We should also be using the same tokensource for `perRPCCredentials` as well. Please also add the test case for that scenario. Same logic but different method call